fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchReverseChainTest.java

```diff
@@ -0,0 +1,1034 @@
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Search parameter 'encounter' target types do not include expected type 'Patient' for '_has' (reverse chain) search.");
+    }
+
+    @Test(groups = { "server-search-reverse-chain" })
+    public void testSearchReverseChainWithSearchParmUndefinedModifierError() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_has:Procedure:subject:code:badModifier", "test")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Undefined Modifier: badModifier");
+    }
+
+    @Test(groups = { "server-search-reverse-chain" })
+    public void testSearchReverseChainWithSearchParmUnsupportedModifierError() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_has:Procedure:subject:code:contains", "test")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Unsupported type/modifier combination: token/contains");
+    }
+
+    @Test(groups = { "server-search-reverse-chain" })
+    public void testSearchReverseChainWithSearchParmIsResultParmError() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_has:Procedure:subject:_total", "1")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Search parameter '_total' for resource type 'Procedure' was not found.");
+    }
+
+    @Test(groups = { "server-search-reverse-chain" })
+    public void testSearchReverseChainWithChainedSearchParmBadModifierError() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_has:Procedure:subject:encounter:contains.status", "test")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Modifier: 'CONTAINS' not allowed on chained parameter");
```

Comment:
Why does this become uppercase, should we upper case it?

Reply:
changed the code in SearchUtil that constructs the exception to use Modifier.value() so we get the natural value